### PR TITLE
APEXCORE-470 New API in DAG - setOperatorAttribute

### PR DESCRIPTION
**Pull request by sandeshh**
1. Added the new API - setOperatorAttribute 
2. updated the unit tests,
3. marked "setAttribute(Operator..."  as deprecated

@PramodSSImmaneni please review.

### CODE REVIEW

File: api/src/main/java/com/datatorrent/api/DAG.java [hunk 1]
```diff
@@ -238,9 +238,15 @@
   /**
    * <p>setAttribute.</p>
    */
+  @Deprecated
   public abstract <T> void setAttribute(Operator operator, Attribute<T> key, T value);
 
   /**
+   * <p>setOperatorAttribute.</p>
```
Comment (pramodin) [inline api/src/main/java/com/datatorrent/api/DAG.java]: javadoc

File: api/src/main/java/com/datatorrent/api/DAG.java [hunk 2]
```diff
@@ -236,11 +236,22 @@
...
   /**
-   * <p>setAttribute.</p>
+   *
+   * Use {@link #setOperatorAttribute} instead
    */
+  @Deprecated
   public abstract <T> void setAttribute(Operator operator, Attribute<T> key, T value);
 
   /**
+   * Set an attribute on the operator.
```
Comment (pramodin) [inline api/src/main/java/com/datatorrent/api/DAG.java]: for an operator

File: api/src/main/java/com/datatorrent/api/DAG.java [hunk 3]
```diff
@@ -236,11 +236,22 @@
...
+   * @param <T> Object type of the attribute.
```
Comment (pramodin) [inline api/src/main/java/com/datatorrent/api/DAG.java]: The value type

File: api/src/main/java/com/datatorrent/api/DAG.java [hunk 4]
```diff
@@ -236,11 +236,22 @@
   public abstract <T> void setAttribute(Attribute<T> key, T value);
 
   /**
-   * <p>setAttribute.</p>
+   *
+   * Use {@link #setOperatorAttribute} instead
    */
```
Comment (pramodin) [inline api/src/main/java/com/datatorrent/api/DAG.java:241]: How about using javadoc deprecation as well
https://docs.oracle.com/javase/tutorial/java/annotations/predefined.html

Comment (sandeshh) [inline api/src/main/java/com/datatorrent/api/DAG.java:241]: Addressed all the review comments. Travis builds fails randomly. I just updated JavaDoc, so it shouldn't have failed the Travis.

**Pull request merged**